### ISSUE TITLE
Build of nordugrid-arc failed with SWIG 4.2.0

### ISSUE THREAD

**Issue by jplesnik** [swig/swig#2745]
I am testing rebuild of SWIG dependencies against version 4.2.0 in Fedora rawhide/40.

The build of `nordugrid-arc-6.18.0` is failing with this version. 

One of the tests is failing:

```
FAIL: ServiceEndpointRetrieverTest.py
===============================================
1 of 9 tests failed
Please report to http://bugzilla.nordugrid.org/
===============================================
...
EEEEEEEEEEEEEEEEE
======================================================================
ERROR: test_constructor_returns_immediately (__main__.ServiceEndpointRetrieverTest.test_constructor_returns_immediately)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/nordugrid-arc-6.18.0/python/test/python/./ServiceEndpointRetrieverTest.py", line 9, in setUp
    arc.ServiceEndpointRetrieverPluginTESTControl.condition.clear()
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'SwigPyObject' object has no attribute 'clear'

======================================================================
ERROR: test_deleting_the_consumer_before_the_retriever (__main__.ServiceEndpointRetrieverTest.test_deleting_the_consumer_before_the_retriever)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/nordugrid-arc-6.18.0/python/test/python/./ServiceEndpointRetrieverTest.py", line 9, in setUp
    arc.ServiceEndpointRetrieverPluginTESTControl.condition.clear()
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'SwigPyObject' object has no attribute 'clear'

```

Complete build log is [here](https://download.copr.fedorainfracloud.org/results/jplesnik/swig-rebuild/fedora-rawhide-x86_64/06843367-nordugrid-arc/builder-live.log.gz). Other information about the build can be found [here](https://copr.fedorainfracloud.org/coprs/jplesnik/swig-rebuild/build/6843367/).

**Comment by cbldlpl** [comment]
I will minimize this issue and update later

**Comment by wsfulton** [comment]
This is due to e3ccabbd4dc22a7e59a9d1b93308d96a97d35328. 

I managed to create a cutdown testcase which involves accessing a static member `std::list<SimpleCondition*>` in `Arc::ServiceEndpointRetrieverPluginTESTControl` where the container is instantiated in another module, but the %import is not done correctly and is as good as missing. However, when corrected the type information is then good, but it exposes what seems to be a long standing quirk in wrapping static member variables in that they are accessed by value rather than by pointer (unlike non-static member variables). The test code fails because it wants to access the `std::list<SimpleCondition*>::clear()` method but the marshalling returns a Python tuple by value instead.

Am still pondering what to suggest, but it might be good if one of the nordugrid developers were available to discuss alternative APIs.

**Comment by wsfulton** [comment]
@jplesnik I've come up with a solution so that nordugrid-arc works with all versions of swig. 

747c3ff18ab7299dadc98c821d6ac5a5470cf3f8 contains a fix for correctly wrapping static member variables that are STL containers, so that the `clear()` method is available. However, it also requires a change in the nordugrid-arc code base. I've requested an account at https://source.coderefinery.org/nordugrid/arc/ in order to submit a patch to them, but don't have it yet. Meanwhile, could you please apply these two patches for testing in Fedora:

[0001-Restore-ServiceEndpointRetrieverTest.py-test-for-swi.patch.txt](https://github.com/swig/swig/files/14322892/0001-Restore-ServiceEndpointRetrieverTest.py-test-for-swi.patch.txt)
[0002-Restore-support-for-swig-4.2.0-and-earlier.patch.txt](https://github.com/swig/swig/files/14322893/0002-Restore-support-for-swig-4.2.0-and-earlier.patch.txt)

The 1st patch fixes it for swig-4.2.1 and the second then allows all versions of SWIG to work again. 

Can you also please remove the patch https://source.coderefinery.org/nordugrid/arc/-/commit/01275ab1f331da703b04637cf39185a1185336cd that you have called 0001-Disable-ServiceEndpointRetrieverTest.py-not-compatib.patch that I see being applied in https://download.copr.fedorainfracloud.org/results/jplesnik/swig-rebuild/fedora-rawhide-x86_64/07019129-nordugrid-arc/builder-live.log.gz. This will then re-enable the failing ServiceEndpointRetrieverTest.py test in nordugrid-carc. The Fedora build should then pass, then we can close this ticket.

**Comment by wsfulton** [comment]
Excellent, thanks.
